### DISCUSSION TITLE
About Bytebuf

### DISCUSSION THREAD

**Discussion by zhangtianxiao** [activej/activej#229] [Q&A] [answered]
ByteBuf is a part of HttpResponse.
To avoid copying, I must implement json library based ByteBuf but not byte[].
Because of I don't know when jsonlib can be reused.
This is a worry.
```java
 class JsonLib{
   byte[] bytes[]; int offset; int length;
 }
 public static ByteBuf toByteBuf(JsonLib json){
   return ByteBuf.wrapForRead(json.bytes,json.offset,json.length);
 }
 var servlet = req->{
 -- The next request will also use jsonlib, and confusion will occu
  var jsonLib  = Pooled.getJsonLib();
   return HttpResponse.ok200().withBody(jsonLib.toByteBuf());
 }
```

**Comment by eduard-vasinskyi** [comment, marked as answer]
Hi, @zhangtianxiao

I think you still have to copy byte arrays, even if `HttpResponse` will use `byte[]` instead of `ByteBuf`. If you set a byte array from your JSON lib as a body of `HttpResponse` and then the next request is received and handled before the previous response is sent to the network, you could alter the contents of the byte array, which will cause inconsistency.
